open-build-service/notqmail-rpmlintrc

```diff
@@ -0,0 +1,14 @@
+addFilter(".* unversioned-explicit-self-provides")
+addFilter(".* non-standard-gid")
+addFilter(".* is not allowed anymore in FHS .*")
+addFilter(".* invalid-license .*")
```

Comment:
These macros supress rpmlint errors. On SLES, it is impossibel to build without supressing these errors. Lot of these are noise because qmail does things different than standard linux distributions. Let me explain each of those lines

# addFilter(".* unversioned-explicit-self-provides")
The unversioned-explicit-self provides can be removed. That error happens when we provide a library and say
```
Provides: qmail-libs
```
RPM and debian generates automatically any shared library provided by package. But sometimes you want your own name to be provided. e.g. MySQL provides mysql-libs but the actual library name provided is libmysqlclient.so.x.x

# addFilter(".* non-standard-gid")

When we install packages the binaries go into /var/qmail/bin with the group as `qmail`. qmail-queue gets installed with owner as `qmailq:qmail`. At the moment none of the current linux distributions will recognize these uids to be valid for binaries outside the user's home directories

# addFilter(".* is not allowed anymore in FHS .*")

This is absolutely needed on the open build service because putting binaries in /var/qmail/bin goes against FHS. This error is printed for each and every binary put in /var/qmail/bin

# addFilter(".* invalid-license .*")

I'm not sure from which file the open build service rpmlint program picks up the list of valid licenses. In our case we have the following licence which the build service complains

```
License: CC-PDDC
```

Note: I have removed `addFilter(".* unversioned-explicit-self-provides")` from the rpmlintrc file and will be part of the next commit